shared/fs/common/path-item-action/menu.tsx

```diff
@@ -192,16 +191,18 @@ const makeMenuItems = (props: Props, hideMenu: () => void) => {
 export default (props: Props) => {
   Kbfs.useFsFileContext(props.path)
   const {downloadID, downloadIntent} = Container.useSelector(state => state.fs.pathItemActionMenu)
-  Kbfs.useFsWatchDownloadForMobile(downloadID || '', downloadIntent)
+  const justDoneWithIntent = Kbfs.useFsWatchDownloadForMobileReturnTrueIfJustDoneWithIntent(
+    downloadID || '',
+    downloadIntent
+  )
```

Comment:
Is this going to work? I.e. update the variable and re-call this code?

Reply:
Not sure I understand the question correctly, but there’s useSelector inside this hook, which triggers the renders. It works kind of like a useState except connected to the redux store. The returned value doesn’t drive re-render directly, but tells this component whether to call `hide()`.